File: hackerrank/java/src/CompareTriplets.java
import java.util.ArrayList;
import java.util.Iterator;
import java.util.List;

// https://www.hackerrank.com/challenges/compare-the-triplets
public class CompareTriplets {
    public static List<Integer> compareWithIterator(List<Integer> a, List<Integer> b) {
        int aliceScore = 0;
        int bobScore = 0;

        Iterator<Integer> aScores = a.iterator();
        Iterator<Integer> bScores = b.iterator();

        List<Integer> answ = new ArrayList<>();

        while(aScores.hasNext() && bScores.hasNext()){
            Integer ai = (Integer) aScores.next();
            Integer bi = (Integer) bScores.next();

            if(ai.compareTo(bi) > 0){
                aliceScore++;
            }else if (ai.compareTo(bi) < 0){
                bobScore++;
            }

        }

        answ.add(aliceScore);
        answ.add(bobScore);
        return answ;
    }
}

File: hackerrank/python/src/compare_triplets.py
#https://www.hackerrank.com/challenges/compare-the-triplets/problem
from typing import List

class CompareTriplets:
    def compare_as_lists(self, a, b):
        print("compar_as_lists",a,b)
        alice_score = 0
        bob_score = 0
        for i in range(len(a)):
            if(a[i] > b[i]):
                alice_score += 1
            elif (a[i] < b[i]):
                bob_score += 1

        return [alice_score,bob_score]



test = CompareTriplets()
res = test.compare_as_lists([1, 2, 3], [3, 2, 1])
print(res)
